mlir/lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -632,8 +632,12 @@ llvm::Constant *mlir::LLVM::detail::getLLVMConstant(
           llvm::ElementCount::get(numElements, /*Scalable=*/isScalable), child);
     if (llvmType->isArrayTy()) {
       auto *arrayType = llvm::ArrayType::get(elementType, numElements);
-      SmallVector<llvm::Constant *, 8> constants(numElements, child);
-      return llvm::ConstantArray::get(arrayType, constants);
+      if (child->isZeroValue()) {
+        return llvm::ConstantAggregateZero::get(arrayType);
+      } else {
+        std::vector<llvm::Constant *> constants(numElements, child);
```

Comment:
Comment added. 

Do we want to have an if to switch between SmallVector and std::vector?

Reply:
Are you sure SmallVector has a size limitation that is different from `std::vector`? It has a fixed stack size but you can resize using a size_t value so I would expect 2^32 elements should be fine. I would drop the 8 that was there before that seems a bit arbitrary and just use `SmallVector<llvm::Constant *> constants(numElements, child);`.

Reply:
Let me try that.

Reply:
Size seems to be limited to `4294967295 `

```
SmallVector unable to grow. Requested capacity (4294967296) is larger than maximum value for size type (4294967295)
```

Reply:
I pasted the documentation of small vector in this PR already: https://github.com/llvm/llvm-project/pull/93230#discussion_r1612337183

Pasting here:

```
/// The template parameter specifies the type which should be used to hold the
/// Size and Capacity of the SmallVector, so it can be adjusted.
/// Using 32 bit size is desirable to shrink the size of the SmallVector.
/// Using 64 bit size is desirable for cases like SmallVector<char>, where a
/// 32 bit size would limit the vector to ~4GB. SmallVectors are used for
/// buffering bitcode output - which can exceed 4GB.
```

And the size is picked with:

```
template <class T>
using SmallVectorSizeType =
    std::conditional_t<sizeof(T) < 4 && sizeof(void *) >= 8, uint64_t,
                       uint32_t>;
```

Reply:
Didn't see this comment. Let's use std:vector then!